### PR TITLE
Add String Search Algorithms in Java & C++

### PR DESCRIPTION
**Pull request by Amisha-Mohapatra**
String searching algorithms in java and c++.

### CODE REVIEW

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
files checked 
- [x] Readme.md
- [x] kmp.cpp
- [x] rabin-carp.cpp
- [ ] kmp.java
- [ ] rabin-carp.java

main issue is only with `readme.md` and some minor issues in `kmp.cpp` (as mentioned in comments )

I think I am not qualified enough to review java code as java is not my area of expertise, but functionally they are working for the cases provided and the ones on which I have tested cpp ones.

File: strings/README.md [hunk 1]
```diff
@@ -18,3 +18,16 @@ You could use any online IDE (for an example [.net Finddle](https://dotnetfiddle
 
 1. [Palindrome Check](java/palindrome.java)
 2. [All subsequences](java/sequence.java)
+
+# String algorithms
```
Comment (Arsenic-ATG) [inline strings/README.md:22]: actually, this title looks better than `String operations` 

here is the proposed change 

change the title from `#String operations` to `#String algorithms`

and then merge your algorithms with rest of them, something like this 

```md
### C or C++
1. [Palindrome Check](c-or-cpp/palindrome.c)
2. [All subsequences](c-or-cpp/sequence.cpp)
3. [KMP String Searching](c-or-cpp/kmp.cpp)
4. [Rabin Karp String Searching](c-or-cpp/rabin-karp.cpp)
```

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
about readme.md ( somehow missed in my previous review)

**Comment by Amisha-Mohapatra** [comment]
@Arsenic-ATG kindly review changes

**Comment by paulsonjpaul** [comment]
@Amisha-Mohapatra, please always make your issues and pull requests' titles descriptive. `Add algorithms` doesn't make any sense as this project is full of algorithms. I have changed it for now.

**Comment by Amisha-Mohapatra** [comment]
> @Amisha-Mohapatra, please always make your issues and pull requests' titles descriptive. `Add algorithms` doesn't make any sense as this project is full of algorithms. I have changed it for now.

@paulsonjpaul I will keep this in mind.

**Review by Arsenic-ATG — APPROVED** [review]

**Pull request merged**